src/palace/manager/sqlalchemy/model/work.py

```diff
@@ -1707,7 +1710,18 @@ def _set_value(parent, key, target):
                     lc["medium"] = doc.presentation_edition.medium
                 lc["licensepool_id"] = license_pool.id
                 lc["quality"] = doc.quality
+                collection_settings = (
+                    license_pool.collection.integration_configuration.settings_dict
+                )
+                lc["lane_priority_level"] = collection_settings.get(
+                    "lane_priority_level",
+                    IntegrationConfigurationConstants.DEFAULT_LANE_PRIORITY_LEVEL,
+                )
                 result["licensepools"].append(lc)
+            # use the maximum lane priority level associated with the work.
+            result["lane_priority_level"] = max(
+                [lc["lane_priority_level"] for lc in result["licensepools"]]
+            )
```

Comment:
Change requested: I might've missed something, but I didn't see anything testing this max lane priority level part. We could test that by adding a book to `test_lane_priority_level_ordering` that is in both collection 1 and collection 3 and making sure that the book appears in the priority collection in both scenarios.

Reply:
Excellent catch - I'll add a test for that.